src/linking/static_lifetime_init.cpp

```diff
@@ -77,14 +69,18 @@ static_lifetime_init(const irep_idt &identifier, symbol_tablet &symbol_table)
     return {}; // do not initialize
   }
 
-  if(symbol.value.id() == ID_nondet)
-  {
-    return {}; // do not initialize
-  }
-
   exprt rhs;
 
-  if(symbol.value.is_nil())
+  if((symbol.value.is_nil() && symbol.is_extern) ||
```

Comment:
Does this also cover `argc` and `argv`?

Reply:
Checked: no, they are initialised by dangling globals `argc'` and `argv'` created in `ansi_c_entry_point`, *after* `static_lifetime_init` has been called. I tried initialising them explicitly: this worked fine except for test `goto-instrument/harness1`, which tries to dump a C harness to replicate a particular situation. That led to a line like `argv' = nondet_0();`, which is not legal C.

So it's not trivial to fix this, but I think we're safe: there's no way for `argc'` and similar to be assigned, so they should never be merged. I agree it would be better to explicitly init them, but given the rabbit hole we appear to be diving down here I'm inclined to leave it to a subsequent PR.